File: benchmarks/json/json.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
)

type Item struct {
	ID     int      `json:"id"`
	Name   string   `json:"name"`
	Price  float64  `json:"price"`
	Tags   []string `json:"tags"`
	Active bool     `json:"active"`
}

type Doc struct {
	Items []Item `json:"items"`
}

func main() {
	if len(os.Args) < 2 {
		os.Exit(2)
	}
	data, err := os.ReadFile(os.Args[1])
	if err != nil {
		os.Exit(1)
	}
	var d Doc
	if err := json.Unmarshal(data, &d); err != nil {
		os.Exit(1)
	}
	sum := 0.0
	active := 0
	for _, it := range d.Items {
		sum += it.Price
		if it.Active {
			active++
		}
	}
	fmt.Printf("items=%d sum=%.2f active=%d\n", len(d.Items), sum, active)
}

File: benchmarks/json/json.c
// JSON parse + walk via yyjson
#include <stdio.h>
#include <stdlib.h>
#include <yyjson.h>

int main(int argc, char** argv) {
    if (argc < 2) { fprintf(stderr, "usage: json <file>\n"); return 2; }
    yyjson_doc* doc = yyjson_read_file(argv[1], 0, NULL, NULL);
    if (!doc) { fprintf(stderr, "parse failed\n"); return 1; }
    yyjson_val* root = yyjson_doc_get_root(doc);
    yyjson_val* items = yyjson_obj_get(root, "items");
    double sum_price = 0;
    long active_count = 0;
    size_t idx, max;
    yyjson_val* item;
    yyjson_arr_foreach(items, idx, max, item) {
        sum_price += yyjson_get_real(yyjson_obj_get(item, "price"));
        if (yyjson_get_bool(yyjson_obj_get(item, "active"))) active_count++;
    }
    printf("items=%zu sum=%.2f active=%ld\n", max, sum_price, active_count);
    yyjson_doc_free(doc);
    return 0;
}
